msxsecurity.go

```diff
@@ -48,12 +49,17 @@ type MsxSecurity struct {
 
 // HasPermission will return true or false given an HTTP request and target permission.
 func (m *MsxSecurity) HasPermission(r *http.Request, perm string) (bool, User) {
-	token := r.Header.Get("Authorization")
-	if len(token) == 0 {
+	authHeader := r.Header.Get("Authorization")
+	if len(authHeader) == 0 {
 		return false, User{}
 	}
-	token = strings.Split(token, " ")[1]
-	return m.checkToken(token, perm)
+	//follows "Bearer tokenvalue" pattern
+	authorization := strings.Split(authHeader, " ")
```

Comment:
The jwt header is in the form `Bearer <token>`; if we match explicitly match the header part, it would be cleaner.

```
	const authHeaderPrefix = "Bearer "
	token = strings.Replace(header, authHeaderPrefix, "", -1))
```

Note: I haven’t tested the code.

Reply:
We just need the second part, the reason I added the length check is to safe check before accessing the index value

Reply:
noted. than makes sense. approving it.